src/Domain/CoordinateMaps/Wedge3D.cpp

```diff
@@ -0,0 +1,357 @@
+         "Inner sphericity must be between 0 and 1");
+  ASSERT(radius_of_other_surface < radius_of_spherical_surface or
+             (1 + sphericity_of_other_surface * (sqrt(3.) - 1.)) *
+                     radius_of_other_surface >
+                 sqrt(3.) * radius_of_spherical_surface,
+         "For the value of the given radii and sphericity, the spherical "
+         "surface intersects with the other surface");
+}
+
+template <typename T>
+std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, 3> Wedge3D::
+operator()(const std::array<T, 3>& x) const noexcept {
+  using ReturnType = std::decay_t<tt::remove_reference_wrapper_t<T>>;
+
+  const ReturnType& xi = x[0];
+  const ReturnType& eta = x[1];
+  const ReturnType& zeta = x[2];
+  ReturnType x_inner_cubical_face = xi * radius_of_other_surface_ / sqrt(3.0);
+  ReturnType y_inner_cubical_face = eta * radius_of_other_surface_ / sqrt(3.0);
+  auto z_inner_cubical_face =
+      make_with_value<ReturnType>(xi, radius_of_other_surface_ / sqrt(3.0));
+  const ReturnType x_unit_spherical_face = tan(M_PI_4 * xi);
+  const ReturnType y_unit_spherical_face = tan(M_PI_4 * eta);
+  const auto z_spherical_faces_common_factor =
+      make_with_value<ReturnType>(xi, 1.) /
+      sqrt(1.0 + square(x_unit_spherical_face) + square(y_unit_spherical_face));
+  const ReturnType z_inner_spherical_face =
+      radius_of_other_surface_ * z_spherical_faces_common_factor;
+
+  // If sphericity_of_other_surface_==0, the inner_cubical_face variables
+  // correspond to the
+  // face of a cube.
+  // If sphericity_of_other_surface_!=0, the inner_cubical_face variables have
+  // the
+  // spherical_face portions added to them to create a rounded-out cube face.
+  // At sphericity_of_other_surface_==1, the face becomes spherical.
+
+  x_inner_cubical_face +=
+      sphericity_of_other_surface_ *
+      (z_inner_spherical_face * x_unit_spherical_face - x_inner_cubical_face);
+  y_inner_cubical_face +=
+      sphericity_of_other_surface_ *
+      (z_inner_spherical_face * y_unit_spherical_face - y_inner_cubical_face);
+  z_inner_cubical_face += sphericity_of_other_surface_ *
+                          (z_inner_spherical_face - z_inner_cubical_face);
+
+  const ReturnType z_outer_spherical_face =
+      radius_of_spherical_surface_ * z_spherical_faces_common_factor;
+  const ReturnType x_outer_spherical_face =
+      z_outer_spherical_face * x_unit_spherical_face;
+  const ReturnType y_outer_spherical_face =
+      z_outer_spherical_face * y_unit_spherical_face;
+  const ReturnType physical_x =
+      x_inner_cubical_face +
+      (x_outer_spherical_face - x_inner_cubical_face) * (zeta + 1) * 0.5;
+  const ReturnType physical_y =
+      y_inner_cubical_face +
+      (y_outer_spherical_face - y_inner_cubical_face) * (zeta + 1) * 0.5;
+  const ReturnType physical_z =
+      z_inner_cubical_face +
+      (z_outer_spherical_face - z_inner_cubical_face) * (zeta + 1) * 0.5;
```

Comment:
I am confused by the variable names in the previous  ~40 lines of code.

For example, it looks like `physical_x` is obtained by interpolating in zeta between `x_inner_cubical_face` and `x_outer_spherical_surface`. But what if the inner surface is the spherical one, and the outer surface is the cubical one? What if both are spherical? What if one is of intermediate curvature?

It is not clear to me how the variables in this function are related to the "spherical surface, other surface" interface of the class. I would like to see improved names, but there should at least be comments.

Reply:
I think @gsb76 is waiting for this to be merged and is going to go through and clean up some of this and come up with clearer variable names since he's figured out some simplifications that can be made.